templates/demo-store/app/components/SortFilter.tsx

```diff
@@ -301,35 +278,22 @@ function PriceRangeFilter({max, min}: {max?: number; min?: number}) {
 }
 
 function filterInputToParams(
-  type: FilterType,
-  rawInput: string | Record<string, any>,
+  rawInput: string | ProductFilter,
   params: URLSearchParams,
 ) {
-  const input = (
-    typeof rawInput === 'string' ? JSON.parse(rawInput) : rawInput
-  ) as Record<string, any>;
-  switch (type) {
-    case 'PRICE_RANGE':
-      if (input.price.min) params.set('minPrice', input.price.min);
-      if (input.price.max) params.set('maxPrice', input.price.max);
-      break;
-    case 'LIST':
-      Object.entries(input).forEach(([key, value]) => {
-        if (typeof value === 'string') {
-          params.set(key, value);
-        } else if (typeof value === 'boolean') {
-          params.set(key, value.toString());
-        } else {
-          const {name, value: val} = value as {name: string; value: string};
-          const allVariants = params.getAll(`variantOption`);
-          const newVariant = `${name}:${val}`;
-          if (!allVariants.includes(newVariant)) {
-            params.append('variantOption', newVariant);
-          }
-        }
-      });
-      break;
-  }
```

Comment:
The changes to `filterInputToParams` are at the heart of this PR!
Previously we were trying to craft URL params based on the type of filter being used, and may have come up with key-value pairs like
```
productVendor=Snowdevil
productType=Snowboards
available=true
minPrice=50
variantOption=material%3ACarbon-fiber
```
Which would then have to be [parsed](https://github.com/Shopify/hydrogen/pull/1526/files#diff-99791b90d83548a12c3750bbbc5386f24808dcad5ac8f7408b45ec85bda120d0L62) based on the key content into parameters for the [`filters` input for the storefronts query ](https://github.com/Shopify/hydrogen/pull/1526/files#diff-99791b90d83548a12c3750bbbc5386f24808dcad5ac8f7408b45ec85bda120d0L106)

Now, we try to use the `input` for filter values directly as we get it from the backend.  This is pretty much already in a JSON format that we can easily use in URL params, then parse out and use for the filters input for the storefronts query. Not only does this clean up all the [special handling](https://github.com/Shopify/hydrogen/pull/1526/files#diff-99791b90d83548a12c3750bbbc5386f24808dcad5ac8f7408b45ec85bda120d0L62) of different keys, but this will also support filtering with metafields/metaobjects without even having to mention them 😄